src/ch/imedias/rsccfx/view/RsccSupportPresenter.java

```diff
@@ -61,45 +58,54 @@ public void initSize(Scene scene) {
   }
 
   /**
-   * Validates the token and displays a symbolic image.
-   *
-   * @param token the token to be validated.
-   * @return path to the image to display.
+   * Sets the token validation image depending on the validity of the token.
    */
-  public Image validationImage(String token) {
-
-    if (validateToken(token)) {
-      view.connectBtn.setDisable(false);
-      return new Image(getClass().getClassLoader().getResource("emblem-default.png")
-          .toExternalForm());
-    }
-    view.connectBtn.setDisable(true);
-    return new Image(getClass().getClassLoader().getResource("dialog-error.png")
-        .toExternalForm());
+  public void setValidationImage(boolean isValid) {
+    String imageFileName = isValid ? VALID_IMAGE_FILENAME : INVALID_IMAGE_FILENAME;
+    view.isValidImg.setImage(
+        new Image(getClass().getClassLoader().getResource(imageFileName).toExternalForm())
+    );
   }
 
   /**
    * Updates the validation image after every key pressed.
    */
   private void attachEvents() {
-
-    view.tokenFld.setOnKeyReleased(event -> {
-      view.isValidImg.setImage(validationImage(view.tokenFld.getText()));
-    });
-
+    view.connectBtn.disableProperty().addListener(
+        (observable, oldValue, newValue) -> {
+          // value = whether the button is disabled or not
+          setValidationImage(!newValue);
+        }
+    );
 
     view.connectBtn.setOnAction(event -> {
       model.setKey(view.tokenFld.getText());
       model.connectToUser();
     });
 
-
     // Closes the other TitledPane so that just one TitledPane is shown on the screen.
     view.keyInputPane.setOnMouseClicked(event -> view.predefinedAdressesPane.setExpanded(false));
     view.predefinedAdressesPane.setOnMouseClicked(event -> view.keyInputPane.setExpanded(false));
   }
 
-
+  private void initBindings() {
+    // disable connect button if key is not valid
+    view.connectBtn.disableProperty().bind(
+        Bindings.when(
+            view.tokenFld.textProperty()
```

Comment:
The whole code inside here might be easier in a seperate method like validate. It would allow to use the validation from other places too, maybe after some values in settings got changed or before pressing a button. For me it would also be easier to read